### PR TITLE
String refinement: Remove generator parameter where unused [blocks: #2310]

### PR DESCRIPTION
**Pull request by tautschnig**
The generator is no longer used in the implementation, and thus does not need to
be passed around. Also update the documentation.

@romainbrenguier Existing PRs of yours might make this one redundant, happy to close it and wait for them to be merged.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/solvers/refinement/string_refinement.cpp [hunk 1]
```diff
@@ -1882,12 +1880,11 @@ static exprt instantiate(
 /// entailed by a (transformed) axiom.
 /// \param [in] axiom: the axiom to instantiate
 /// \param index_set: set of indexes
-/// \param generator: constraint generator object
+/// \param witnesses: ???
```
Comment (smowton) [inline src/solvers/refinement/string_refinement.cpp]: @romainbrenguier @allredj provide a description for `witnesses`?

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 266dc5af31c1856a5a73b4163a8ee6390ef433bb).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90995408

File: src/solvers/refinement/string_refinement.cpp (hunk 1 above)
Comment (allredj) [reply src/solvers/refinement/string_refinement.cpp]:
```
`axiom`'s witnesses for non containment
```

Comment (tautschnig) [reply src/solvers/refinement/string_refinement.cpp]: Thanks, added!

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: ec2216831373bb2260f01ba44a03fd10148e6828).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91044356

**Review by romainbrenguier — APPROVED** [review]

**Review by smowton — APPROVED** [review]

**Pull request merged**